components/eamxx/src/physics/cosp/cosp_c2f.F90

```diff
@@ -292,23 +292,44 @@ subroutine cosp_c2f_run(npoints, ncolumns, nlevels, ntau, nctp, &
  
     nptsperit = npoints
 
-    tca(:npoints,:nlevels) = cldfrac(:npoints,:nlevels)
-    cca(:npoints,:nlevels) = 0
-    mr_lsliq(:npoints,:nlevels) = 0
-    mr_ccliq(:npoints,:nlevels) = 0
-    mr_lsice(:npoints,:nlevels) = 0
-    mr_ccice(:npoints,:nlevels) = 0
-    dtau_c(:npoints,:nlevels) = 0
-    dtau_s(:npoints,:nlevels) = dtau067(:npoints,:nlevels)
-    dem_c (:npoints,:nlevels) = 0
-    dem_s (:npoints,:nlevels) = 1._wp - exp(-dtau105(:npoints,:nlevels))
-    mr_lsrain(:npoints,:nlevels) = 0
-    mr_ccrain(:npoints,:nlevels) = 0
-    mr_lssnow(:npoints,:nlevels) = 0
-    mr_lssnow(:npoints,:nlevels) = 0
-    mr_ccsnow(:npoints,:nlevels) = 0
-    mr_lsgrpl(:npoints,:nlevels) = 0
-    reff = 0  ! FIXME
+    ! In-cloud values are assumed. If ncolumns = 1, then convert in-cloud values to gridbox
+    if (ncolumns == 1) then
```

Comment:
Up to you, but since only 2 arrays are treated differently, you may just do all the `=0` assignments outside the if, and leave only `dtau_s` and `dem_s` in the conditional.

Reply:
I left the rest here because in the near future these will not be all zero, and wanted to keep all of the like fields together to organize things. @whannah1 brought this up too.

Reply:
Decided to fix this anyways. Pulled the unused fields out of the logical and added doc about all of them.